dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1678,21 +1702,54 @@ std::tuple<winrt::Rect, winrt::Thickness, winrt::Thickness> TeachingTip::Determi
     }
 }
 
-std::array<winrt::TeachingTipPlacementMode, 12> TeachingTip::GetPlacementFallbackOrder(winrt::TeachingTipPlacementMode preferredPlacement)
+winrt::Rect TeachingTip::GetEffectiveWindowBounds()
+{
+    if (m_useTestWindowBounds)
+    {
+        return m_testWindowBounds;
+    }
+#ifdef USE_INSIDER_SDK
+    if (winrt::IUIElement10 uiElement10 = *this)
+    {
+        if (auto xamlRoot = uiElement10.XamlRoot())
+        {
+            return winrt::Rect{ 0,0,xamlRoot.Size().Width, xamlRoot.Size().Height };
+        }
+    }
+#endif // USE_INSIDER_SDK
+    return winrt::Window::Current().CoreWindow().Bounds();
+}
+
+winrt::Rect TeachingTip::GetEffectiveScreenBounds()
+{
+    if (!m_useTestScreenBounds && !ShouldConstrainToRootBounds())
+    {
+        auto displayInfo = winrt::DisplayInformation::GetForCurrentView();
+        auto scaleFactor = displayInfo.RawPixelsPerViewPixel();
```

Comment:
I keep reading this code and thinking it will come into play -- would you add a MUX_ASSERT_MSG(!m_returnTopForOutOfWindowPlacement) just to make me feel better?